xmodule/capa/safe_exec/safe_exec.py

```diff
@@ -182,17 +187,27 @@ def safe_exec(
         # Run the code in both the remote codejail service as well as the local codejail
         # when in darklaunch mode.
         if is_codejail_in_darklaunch():
-            data = {
-                "code": code_prolog + LAZY_IMPORTS + code,
-                "globals_dict": globals_dict,
-                "python_path": python_path,
-                "limit_overrides_context": limit_overrides_context,
-                "slug": slug,
-                "unsafely": unsafely,
-                "extra_files": extra_files,
-            }
-
-            get_remote_exec(data)
+            try:
+                # Create a copy so the originals are not modified as part of this call.
+                darklaunch_globals = copy.deepcopy(globals_dict)
+                data = {
+                    "code": code_prolog + LAZY_IMPORTS + code,
+                    "globals_dict": darklaunch_globals,
+                    "python_path": python_path,
+                    "limit_overrides_context": limit_overrides_context,
+                    "slug": slug,
+                    "unsafely": unsafely,
+                    "extra_files": extra_files,
+                }
+                remote_emsg, _remote_exception = get_remote_exec(data)
+                log.info(
+                    f"Remote execution in darklaunch mode produces: {darklaunch_globals} or exception: {remote_emsg}"
+                )
+                log.info(f"Local execution in darklaunch mode produces: {globals_dict} or exception: {emsg}")
+            except Exception as e:  # pylint: disable=broad-except
```

Comment:
Nit: You could add the no coverage pragma (whatever it is) if we don't intend to test this case, so it doesn't affect coverage.